src/client.rs

```diff
@@ -965,10 +992,10 @@ impl Client {
     pub async fn pull_blob<T: AsyncWrite + Unpin>(
         &self,
         image: &Reference,
-        layer: &OciDescriptor,
+        layer: &LayerDescriptor<'_>,
```

Comment:
Since `LayerDescriptor` contains purely borrowed types, I think I'd prefer this not also be a borrowed parameter. That would avoid a call that looks like this: `&(&layer).into()` like we have down below

Reply:
Yep, I wondered if that was the case. Let's see if we can figure out the `Deref` thing and then make a call from there